lighthouse-core/scripts/generate-bundlephobia-database.js

```diff
@@ -0,0 +1,178 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+/* eslint-disable no-console */
+
+/** @typedef {import('bundle-phobia-cli').BundlePhobiaLibrary} BundlePhobiaLibrary */
+
+const fs = require('fs');
+const path = require('path');
+const getPackageVersionList = require('bundle-phobia-cli').fetchPackageStats.getPackageVersionList;
+const fetchPackageStats = require('bundle-phobia-cli').fetchPackageStats.fetchPackageStats;
+const databasePath = path.join(__dirname,
+  '../lib/large-javascript-libraries/bundlephobia-database.json');
+
+/** @type {Record<string, string[]>} */
+const suggestionsJSON = require('../lib/large-javascript-libraries/library-suggestions.js')
+  .suggestions;
+
+
+/** @type {string[]} */
+let largeLibraries = [];
+largeLibraries = largeLibraries.concat(...Object.keys(suggestionsJSON));
+
+/** @type {string[]} */
+let suggestedLibraries = [];
+suggestedLibraries = suggestedLibraries.concat(...Object.values(suggestionsJSON));
+
+const totalLibraries = largeLibraries.length + suggestedLibraries.length;
+
+/** @type {Record<string, {lastScraped: number | 'Error', repository: string, versions: any}>} */
+let database = {};
+if (fs.existsSync(databasePath)) {
+  database = require(databasePath);
+}
+
+/**
+ * Returns true if this library has been scraped from BundlePhobia in the past hour.
+ * This is used to rate-limit the number of network requests we make to BundlePhobia.
+ * @param {string} library
+ * @return {boolean}
+ */
+function hasBeenRecentlyScraped(library) {
+  if (!database[library]) return false;
+
+  const lastScraped = database[library].lastScraped;
+  if (!database[library] || lastScraped === 'Error') return false;
+
+  return (Date.now() - lastScraped) / (1000 * 60 * 60) < 1;
+}
+
+/**
+ * Returns true if the object represents valid BundlePhobia JSON.
+ * The version string must not match this false-positive expression: '{number} packages'.
+ * @param {any} library
+ * @return {library is BundlePhobiaLibrary}
+ */
+function validateLibraryObject(library) {
+  return library.hasOwnProperty('name') &&
+    library.hasOwnProperty('size') &&
+    library.hasOwnProperty('gzip') &&
+    library.hasOwnProperty('description') &&
+    library.hasOwnProperty('repository') &&
+    library.hasOwnProperty('version') &&
+    !library.version.match(/^([0-9]+) packages$/);
+}
+
+/**
+ * Save BundlePhobia stats for a given npm library to the database.
+ * @param {string} library
+ * @param {number} index
+ * @param {number} numVersionsToFetchLimit
+ */
+async function collectLibraryStats(library, index, numVersionsToFetchLimit) {
+  return new Promise(async (resolve, reject) => {
```

Comment:
since this function is `async`, no need for wrapping in a `Promise`. Should be able to delete this line, the two `resolve()` lines below, outdent the code, and this should work exactly the same.